### DISCUSSION TITLE
getStaticProps context params not showing custom query param

### DISCUSSION THREAD

**Discussion by Bohmaster** [vercel/next.js#13919] [Help] [answered]
I have this route /pages/propiedades/[id].js with this code:

export async function getStaticProps(context) {
    console.log(1, context.params.name);
    const propiedad = await getPropiedad(context.params.id);

    const filter = {
        where: {
            tipoOperacion: propiedad.tipoOperacion,
            ciudad: propiedad.ciudad,
            tipoPropiedad: propiedad.tipoPropiedad
        }
    }

    const propiedadesParecidas = await getAllPropiedadesParecidas(filter);

    return {
        props: {
            propiedad,
            propiedadesParecidas
        }
    }
}

And a component wit this link

`    <Link href={{ pathname: `/propiedades/${info.id}`, query: { name: 'text' }}} key={info.id}>
`

This link goes to `/propiedades/1?name=text ` but when I log **context.params** in getStaticProps (as above) I only get `{ id: 1 }` missing the **name** param

I'm doing something wrong? I want to get access to the two params (id, name) but only get id.

**Comment by jamesmosier** [comment, marked as answer]
[This answer should be helpful.](https://github.com/vercel/next.js/discussions/10951#discussioncomment-1202)

To quote Tim's response:

> `getStaticPaths` generates static files, which means they can't access querystring at generation time, you can however access them during browser rendering using `useRouter`: https://nextjs.org/docs/api-reference/next/router#userouter

The same is true of `getStaticProps`.

**Comment by Bohmaster** [comment]
Thank you. Now I understand.
